service/controller/app/v1/resource/chartcrd/create.go

```diff
@@ -0,0 +1,61 @@
+package chartcrd
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/controllercontext"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	secretName := key.SecretName(cr)
+
+	if secretName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring chart crd creation on tenant cluster")
+
+		ctlCtx, err := controllercontext.FromContext(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		k8sExtClient, err := apiextensionsclient.NewForConfig(ctlCtx.RESTConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c := k8scrdclient.Config{
+			K8sExtClient: k8sExtClient,
+			Logger:       r.logger,
+		}
+
+		crdClient, err := k8scrdclient.New(c)
+		crdBackoff := backoff.NewMaxRetries(3, 1*time.Second)
+
+		err = crdClient.EnsureCreated(ctx, v1alpha1.NewChartCRD(), crdBackoff)
```

Comment:
When `EnsureCreated` would return an error indicating that the creation got initialized but is not yet completed, we would not need the backoff and could cancel the resource without returning an error. A resource returning an error causes the `retryresource` to execute the resource here again and we start over again, ending up in a wait lock for some time and only throwing errors. I would suggest to use some more specific error handling and cancel the resource gracefully when we notice the CRD installation still takes a bit more time.